MyProject/src/main/java/com/netcracker/controller/InventoryController.java

```diff
@@ -0,0 +1,265 @@
+            @PathVariable String id) {
+
+        return interfaceRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find interface with id: " + id));
+    }
+
+
+    @GetMapping("/hardware-components/{id}")
+    @ApiOperation(value = "Get hardware component by id")
+    public HardwareComponent getHwComponentById(
+            @ApiParam(value = "hardware component id", required = true)
+            @PathVariable String id) {
+
+        return hwRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find hardware component with id: " + id));
+    }
+
+    @GetMapping("/network-elements/{id}")
+    @ApiOperation(value = "Get network element by id")
+    public NetworkElement getNetElementById(
+            @ApiParam(value = "network element id", required = true)
+            @PathVariable String id) {
+
+        return neRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find network element with id: " + id));
+    }
+
+
+
+    @GetMapping("/ne-to-hw/{id}")
+    @ApiOperation(value = "Get ne-to-hw by id")
+    public NeToHw getNeToHwById(
+            @ApiParam(value = "ne-to-hw id", required = true)
+            @PathVariable String id) {
+
+        return neToHwRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find ne-to-hw with id: " + id));
+    }
+
+    @GetMapping("/links/{id}")
+    @ApiOperation(value = "Get link by id")
+    public Link getLinkById(
+            @ApiParam(value = "link id", required = true)
+            @PathVariable String id) {
+
+        return linkRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find link with id: " + id));
+    }
+
+    @PostMapping("/network-elements/{id}/interface")
+    @ApiOperation(value = "Add interface")
+    public Interface createInterface(
+            @ApiParam(value = "network element id", required = true)
+            @PathVariable String id,
+            @ApiParam(value = "interface", required = true)
+            @RequestBody Interface inter ){
+
+        NetworkElement networkElement = neRepository.findById(id).orElseThrow(
+                ()-> new ObjectNotFoundException("Unable to find network element with id: " + id ));
+
+        if(networkElement.getInterfaces().contains(inter)){
+            throw new IllegalStateException("Network element with id: " + id + " already has assigned interface");
+        }
+
+        Interface anInterface = interfaceRepository.save(inter);
+
+        NeToInterface neToInterface = new NeToInterface(networkElement, anInterface);
+
+        neToInterRepository.save(neToInterface);
+
+        return anInterface;
+    }
+
+
+    @PostMapping("/network-elements/{id}/hardware-component")
```

Comment:
I propose to use the foloowing path : "/hw-components/{id}"
where {id} is the is of the parent network-element